### PR TITLE
AWS Get EC2 Instances About To Retired

### PR DESCRIPTION
**Pull request by harshal-netserv**
## Description
AWS Get EC2 Instances About To Retired

### Testing
image

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

**Review by dougsillars — APPROVED** [review]

File: AWS/legos/aws_get_reserved_instances_about_to_retired/aws_get_reserved_instances_about_to_retired.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+##  All rights reserved.
+##
+from pydantic import BaseModel, Field
+from typing import List
+from datetime import datetime, timezone, timedelta
+import pprint
+
+
+class InputSchema(BaseModel):
+    region: str = Field(
```
Comment (shloka-bhalgat-unskript) [inline AWS/legos/aws_get_reserved_instances_about_to_retired/aws_get_reserved_instances_about_to_retired.py:12]: region should be optional

File: AWS/legos/aws_get_reserved_instances_about_to_retired/aws_get_reserved_instances_about_to_retired.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    try:
+        ec2Client = handle.client('ec2', region_name=region)
+        response = ec2Client.describe_reserved_instances()
+        for reserved_id in response["ReservedInstances"]:
+            # check if the Reserved Instance is scheduled to end within the threshold
+            if reserved_id['State'] == 'active' and (reserved_id['End'] - now).days <= threshold:
+                result.append(reserved_id["ReservedInstancesId"])
+    except Exception as e:
+        raise Exception(e)
+    return result
```
Comment (shloka-bhalgat-unskript) [inline AWS/legos/aws_get_reserved_instances_about_to_retired/aws_get_reserved_instances_about_to_retired.py:52]: Check lego format needed here

File: AWS/legos/aws_get_reserved_instances_about_to_retired/aws_get_reserved_instances_about_to_retired.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        description='The threshold for the reserved instance is scheduled to end within the threshold.')
+
+
+def aws_get_reserved_instances_about_to_retired_printer(output):
+    if output is None:
+        return
+    pprint.pprint({"Instances": output})
+
+
+def aws_get_reserved_instances_about_to_retired(handle, region: str = "", threshold: int = 7) -> List:
```
Comment (shloka-bhalgat-unskript) [inline AWS/legos/aws_get_reserved_instances_about_to_retired/aws_get_reserved_instances_about_to_retired.py:28]: This should return a Tuple

**Review by dougsillars — APPROVED** [review]

File: AWS/legos/aws_get_reserved_instances_about_to_retired/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+[<img align="left" src="https://unskript.com/assets/favicon.png" width="100" height="100" style="padding-right: 5px">](https://unskript.com/assets/favicon.png) 
+<h1>AWS Get EC2 Instances About To Retired</h1>
+
+## Description
+This Lego filter AWS reserved instance is scheduled to end within the threshold.
+
+
```
Comment (dougsillars) [inline AWS/legos/aws_get_reserved_instances_about_to_retired/README.md:7]: if the action is a healthcheck - we should say something in the readme

**Review by dougsillars — APPROVED** [review]

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**